### ISSUE TITLE
UV_PYTHON env variable takes precedence over .python-version

### ISSUE THREAD

**Issue by unique1o1** [astral-sh/uv#6645]
When adding a `.python-version` file to a project and afterward using `uv sync`, the UV_PYTHON env variable takes more precedence over `.python-version` and doesn't change to the python version specified inside `.python-version`.

My Question:

Is this the intended behaviour?

UV version: 0.3.3 (deea6025a 2024-08-23)

**Comment by zanieb** [comment]
I think this is intentional. `UV_PYTHON` is equivalent to setting `--python` and both take precedence over `.python-version` files. If you wanted different behavior, I think we'd need to introduce some sort of `UV_DEFAULT_PYTHON` variable.

**Comment by ksamuel** [comment]
I agree. I mistakenly used `UV_PYTHON` to set the default python version when none is passed, and was surprised when running `uv run python` in a project started a different shell version than the one in .python-version.

Having a `UV_DEFAULT_PYTHON` (and signal it exists in the doc  and help message for `UV_PYTHON`) would help avoid this surprise. It's not uncommon to stick to a default version of Python for a while, and update it when needed, so an env var makes for that makes sense.
